server/hotrod/src/main/scala/org/infinispan/server/hotrod/Encoder2x.scala

```diff
@@ -51,7 +49,21 @@ object Encoder2x extends AbstractVersionedEncoder with Constants with Log {
    }
 
    override def writeHeader(r: Response, buf: ByteBuf, addressCache: AddressCache, server: HotRodServer): Unit = {
-      val newTopology = getTopologyResponse(r, addressCache, server)
+      // Sometimes an error happens before we have added the cache to the knownCaches/knownCacheConfigurations map
+      // If that happens, we pretend the cache is LOCAL and we skip the topology update
+      val cr = server.getCacheRegistry(r.cacheName)
+      val configuration = server.getCacheConfiguration(r.cacheName)
+      val cacheMode = configuration match {
+         case null => CacheMode.LOCAL
+         case _ => configuration.clustering().cacheMode()
+      }
```

Comment:
I feel like the extra `Option` and methods would obscure what I'm trying to do...